middleware/realip.go

```diff
@@ -29,7 +30,7 @@ var xRealIP = http.CanonicalHeaderKey("X-Real-IP")
 func RealIP(h http.Handler) http.Handler {
 	fn := func(w http.ResponseWriter, r *http.Request) {
 		if rip := realIP(r); rip != "" {
-			r.RemoteAddr = rip
+			r.RemoteAddr = net.JoinHostPort(rip, "0")
```

Comment:
Why would we add `:0` port to the address? This doesn't make sense to me. Can you clarify?

Reply:
Because that's the format Go uses for RemoteAddr; see  #453

Reply:
I am wondering if the use of port `0` is actually the best way to go? 
Since `chi` is an HTTP router - wouldn't `1.2.3.4` be equal to `1.2.3.4:80` (the default HTTP port)?
Or am I missing something?

Reply:
It's normally set to the source port, not destination port. Any value is "wrong", and at least 0 is "clearly wrong".